refresh.template.py

```diff
@@ -309,7 +308,7 @@ def _get_commands(target: str, flags: str):
         # object_hook allows object.member syntax, just like a proto, while avoiding the protobuf dependency
         parsed_aquery_output = json.loads(aquery_process.stdout, object_hook=lambda d: types.SimpleNamespace(**d))
     except json.JSONDecodeError:
-        print("aquery failed. Command:", shlex.join(cmd), file=sys.stderr)
+        print("aquery failed. Command:", subprocess.list2cmdline(cmd), file=sys.stderr)
```

Comment:
Is it a good idea to call `subprocess.list2cmdline()`?  From this comment https://github.com/python/cpython/blob/14284b0e71c2110f94941b774478fb594d0ef192/Lib/subprocess.py#L65-L66 and https://bugs.python.org/issue10838 it looks like the Python maintainers intend for it to be a private API.

How about this instead?

```python
' '.join(shlex.quote(arg) for arg in cmd)`
```

Reply:
Good catch. Generally I prefer the `subprocess.list2cmdline()` even though that is internal API as I know that is what is used internally and should cope with escaping problems. But this is just for displaying the string to the user. I'll change it.